kong/utils_test.go

```diff
@@ -159,3 +161,356 @@ func Test_requestWithHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestFillRoutesDefaults(T *testing.T) {
+	assert := assert.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.Nil(err)
+	assert.NotNil(client)
+
+	tests := []struct {
+		name     string
+		route    *Route
+		expected *Route
+	}{
+		{
+			name: "fills defaults for all fields except paths, leaves name unchanged",
+			route: &Route{
+				Name:  String("r1"),
+				Paths: []*string{String("/r1")},
+			},
+			expected: &Route{
+				Name:                    String("r1"),
+				Paths:                   []*string{String("/r1")},
+				PreserveHost:            Bool(false),
+				Protocols:               []*string{String("http"), String("https")},
+				RegexPriority:           Int(0),
+				StripPath:               Bool(true),
+				HTTPSRedirectStatusCode: Int(426),
+			},
+		},
+		{
+			name: "fills defaults for all fields except paths and protocols, leaves name unchanged",
+			route: &Route{
+				Name:      String("r1"),
+				Paths:     []*string{String("/r1")},
+				Protocols: []*string{String("grpc")},
+			},
+			expected: &Route{
+				Name:                    String("r1"),
+				Paths:                   []*string{String("/r1")},
+				PreserveHost:            Bool(false),
+				Protocols:               []*string{String("grpc")},
+				RegexPriority:           Int(0),
+				StripPath:               Bool(true),
+				HTTPSRedirectStatusCode: Int(426),
+			},
+		},
+		// TODO: re-add once upstream patch has been merged.
+		// https://github.com/imdario/mergo/pull/203
```

Comment:
It is likely that we will forget about this TODO. Can you remove the code and open a Github issue to track it?

Reply:
https://github.com/Kong/go-kong/issues/122